File: dcp-solutions/2024-12-26/solution.js
```javascript
class TreeNode {
    constructor(val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

function printTreeLevelWise(root) {
    if (!root) return;

    let queue = [root];

    while (queue.length > 0) {
        let levelSize = queue.length;
        for (let i = 0; i < levelSize; i++) {
            let node = queue.shift();
            process.stdout.write(node.val + " ");
            if (node.left) queue.push(node.left);
            if (node.right) queue.push(node.right);
        }
        console.log();
    }
}
```

File: dcp-solutions/2024-12-26/solution.py
```python
from collections import deque

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

def print_level_order(root):
    if not root:
        return

    queue = deque([root])
    while queue:
        level_size = len(queue)
        for _ in range(level_size):
            node = queue.popleft()
            print(node.val, end=' ')

            if node.left:
                queue.append(node.left)
            if node.right:
                queue.append(node.right)

# Test
#      1
#    /   \
#   2     3
#  / \   / \
# 4   5 6   7
root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)
root.left.left = TreeNode(4)
root.left.right = TreeNode(5)
root.right.left = TreeNode(6)
root.right.right = TreeNode(7)

print_level_order(root)  # Output: 1 2 3 4 5 6 7
```